src/libraries/System.Security.Cryptography.Cose/tests/CoseSign1MessageTests.Sign.cs

```diff
@@ -9,60 +10,43 @@
 
 namespace System.Security.Cryptography.Cose.Tests
 {
-    public partial class CoseSign1MessageTests
+    // Tests that apply to all [Try]Sign overloads.
+    public abstract class CoseSign1MessageTests_Sign<T> where T : AsymmetricAlgorithm
     {
-        [Theory]
-        [InlineData(ECDsaAlgorithm.ES256)]
-        [InlineData(ECDsaAlgorithm.ES384)]
-        [InlineData(ECDsaAlgorithm.ES512)]
-        public void SignVerifyECDsa(ECDsaAlgorithm algorithm)
-        {
-            ECDsa ecdsa = ECDsaKeys[algorithm];
-            HashAlgorithmName hashAlgorithm = GetHashAlgorithmNameFromCoseAlgorithm((int)algorithm);
-
-            byte[] coseMessageBytes = CoseSign1Message.Sign(s_sampleContent, ecdsa, hashAlgorithm);
-            CoseSign1Message msg = CoseMessage.DecodeSign1(coseMessageBytes);
-            Assert.True(msg.Verify(ecdsa));
-        }
+        internal CoseAlgorithm DefaultAlgorithm => CoseAlgorithms[0];
+        internal T DefaultKey => GetKeyHashPair<T>(CoseAlgorithms[0]).Key;
+        internal HashAlgorithmName DefaultHash => GetKeyHashPair<T>(CoseAlgorithms[0]).Hash;
+        internal abstract List<CoseAlgorithm> CoseAlgorithms { get; }
+        internal abstract byte[] Sign(byte[] content, T key, HashAlgorithmName hashAlgorithm, bool isDetached = false);
+        internal abstract bool Verify(CoseSign1Message msg, T key);
+        internal abstract bool Verify(CoseSign1Message msg, T key, ReadOnlySpan<byte> content);
 
-        [Theory]
-        [InlineData(RSAAlgorithm.PS256)]
-        [InlineData(RSAAlgorithm.PS384)]
-        [InlineData(RSAAlgorithm.PS512)]
-        public void SignVerifyRSA(RSAAlgorithm algorithm)
+        internal IEnumerable<(T Key, HashAlgorithmName Hash, CoseAlgorithm Algorithm)> GetKeyHashAlgorithmTierce(bool useNonPrivateKey = false)
```

Comment:
I'm not familiar with the word `Tierce`.  I think you're using it as `Triplet`... unless for some reason 35 gallons of wine are involved :smile:

Reply:
Yeah, I was looking for the translation of the spanish word "tercia" and that's what I found in some shady site, I guess it lied to me. 

> Cada jugador anuncia sus combinaciones por el siguiente orden, primero el Brelan, y luego, la Escalera (Quinta, Cuarta o Tercia).
> Each player declares their combinations in the following order, first the Brelan, then the Run (Quint, Quarte or [Tierce](https://context.reverso.net/translation/english-spanish/Tierce)).

Reply:
Mi favorito website para traduzcar (por solo una palabra, vs. frases) esta Wiktionary.

https://en.wiktionary.org/wiki/tercia => https://en.wiktionary.org/wiki/tercio#Spanish => "third" (the ratio of 1/3).  (Pienso que tu (queurer-pasado) "triplete" o "triada", no?  https://en.wiktionary.org/wiki/triplet#Translations)

Es bueno por English a otros lenguas (in cuartos grises/amarillos baja del definicion) y otras lengas a English (como la alta).

(Attempted with no online translation.  Hope it gave you a good laugh.)